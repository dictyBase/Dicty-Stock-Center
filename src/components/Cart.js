// @flow
import React from "react"
import { Link } from "react-router-dom"
import { connect } from "react-redux"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import type { MapStateToProps } from "react-redux"

const styles = theme => ({
  container: {
    margin: "auto",
    width: "75%",

    [theme.breakpoints.up("xl")]: {
      width: "1270px",
    },
  },
  link: {
    color: "#004080",
    textDecoration: "none",
  },
})

type Props = {
  /** List of added items in the cart */
  items: Array<Object>,
  /** Material-UI styling */
  classes: Object,
}

/**
 * This is the cart component that displays between the Navbar and body content on every page.
 * It has a cart icon with the current number of added items next to it.
 */

export const Cart = (props: Props) => {
  const { items, classes } = props

  return (
    <Grid container justify="flex-end" className={classes.container}>
      <Grid item>
        <Link className={classes.link} to="/cart">
          <FontAwesomeIcon icon="shopping-cart" size="2x" /> ({items.length})
        </Link>
      </Grid>
    </Grid>
  )
}

const mapStateToProps: MapStateToProps<*, *, *> = state => ({
  items: state.cart.addedItems,
})

export default connect(mapStateToProps)(withStyles(styles)(Cart))
