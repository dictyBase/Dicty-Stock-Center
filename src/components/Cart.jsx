import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { Flex, Box } from 'rebass'
import { Container } from 'styles'
import FontAwesome from 'react-fontawesome'

export default class Cart extends Component {
    displayName = 'Shopping cart icon'
    static propTypes = {
        cart: PropTypes.object.isRequired
    }
    render() {
        const { addedItems } = this.props.cart
        return (
            <Container>
                <Flex justify="flex-end">
                    <Box>
                        <Link to="/cart">
                            <FontAwesome name="shopping-cart" size="3x" />
                            ({ addedItems.length })
                        </Link>
                    </Box>
                </Flex>
            </Container>
        )
    }
}
