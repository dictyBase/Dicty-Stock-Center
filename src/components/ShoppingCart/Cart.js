// @flow
import React from "react"
import { connect } from "react-redux"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { removeItem } from "actions/cart"
import { Container, TableResponsive, Table, DangerButton } from "styles"

type Props = {
  addedItems: Array<Object>,
  removeItem: Function,
}

const Cart = (props: Props) => {
  const addedItems = props.addedItems
  return (
    <Container>
      <TableResponsive>
        <Table condensed>
          <thead>
            <tr>
              <th>ID</th>
              <th>Strain/Plasmid Name</th>
              <th>Fee</th>
            </tr>
          </thead>
          <tbody>
            {addedItems.map((item, index) => (
              <tr key={index}>
                <td>{item.id}</td>
                <td>{item.name}</td>
                <td>{item.fee}</td>
                <td>
                  <DangerButton
                    type="button"
                    onClick={() => props.removeItem(item.id)}>
                    <FontAwesomeIcon icon="trash" />
                  </DangerButton>
                </td>
              </tr>
            ))}
          </tbody>
        </Table>
      </TableResponsive>
    </Container>
  )
}

const mapStateToProps = state => ({
  addedItems: state.cart.addedItems,
})

const mapDispatchToProps = dispatch => ({
  removeItem: id => {
    dispatch(removeItem(id))
  },
})

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Cart)
