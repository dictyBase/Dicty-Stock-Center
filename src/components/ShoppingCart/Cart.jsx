import React, { Component, PropTypes } from 'react'
import 'styles/core.scss'

export default class Cart extends Component {
    displayName = 'Shopping cart'
    static propTypes = {
        cart: PropTypes.object.isRequired,
        cartActions: PropTypes.object.isRequired
    }

    render() {
        const { cart, cartActions } = this.props
        return (
                <div className="container">
                <div className="table-responsive">
                    <table className="table table-condensed">
                        <thead>
                          <tr>
                            <th>ID</th>
                            <th>Strain/Plasmid Name</th>
                            <th>Fee</th>
                          </tr>
                        </thead>
                        <tbody>
                            {
                                cart.addedItems.map((item, index) => {
                                    return (
                                        <tr key={ index }>
                                          <td>{ item.id }</td>
                                          <td>{ item.name }</td>
                                          <td>{ item.fee }</td>
                                          <td>
                                            <button type="button"
                                              className="btn btn-danger"
                                              onClick={ () => cartActions.removeItem(item.id) }>
                                              <i className="fa fa-trash-o"></i>
                                            </button>
                                          </td>
                                        </tr>
                                    )
                                })
                            }
                        </tbody>
                      </table>
                </div>
                </div>
        )
    }
}
