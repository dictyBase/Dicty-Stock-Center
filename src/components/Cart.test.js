import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Cart from './Cart'

test('matching a snapshot of Cart', () => {
    const cart = {
        addedItems: []
    }

    const component = renderer.create(
        <BrowserRouter>
            <Cart cart={ cart } />
        </BrowserRouter>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
