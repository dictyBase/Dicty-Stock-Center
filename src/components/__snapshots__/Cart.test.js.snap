// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Cart 1`] = `
.c0 {
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

@media (min-width:768px) {
  .c0 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .c0 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .c0 {
    width: 1170px;
  }
}

<div
  className="c0"
>
  <div
    className="sc-htpNat cPcSwG sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="iHZvIS"
      is={null}
    >
      <a
        href="/cart"
        onClick={[Function]}
      >
        <span
          aria-hidden={true}
          className="fa fa-shopping-cart fa-3x"
        />
        (
        0
        )
      </a>
    </div>
  </div>
</div>
`;
